### PR TITLE
Create dumps v3

### PR DESCRIPTION
**Pull request by MarinPostma**
The introduction of the obkv document format has changed the format of the updates, by removing the need for the document format of the addition (it is not necessary since update are store in the obkv format). This has caused breakage in the dumps that this pr solves by introducing a 3rd version of the dumps.

A v2 compat layer has been created that support the import of v2 dumps into meilisearch. This has permitted to move the compat code that existed elsewhere in meiliearch to be moved into the v2 module. The asc/desc patching is now only done for forward compatibility when loading a v2 dump, and the v3 write the asc/desc to the dump with the new syntax.

### CODE REVIEW

File: meilisearch-lib/src/index_controller/dump_actor/loaders/mod.rs [hunk 1]
```diff
@@ -1,2 +1,19 @@
...
+
+mod compat {
+    /// Parses the v1 version of the Asc ranking rules `asc(price)`and returns the field name.
+    pub fn asc_ranking_rule(text: &str) -> Option<&str> {
+        text.split_once("asc(")
+            .and_then(|(_, tail)| tail.rsplit_once(")"))
+            .map(|(field, _)| field)
+    }
+
+    /// Parses the v1 version of the Desc ranking rules `asc(price)`and returns the field name.
```
Comment (irevoire) [inline meilisearch-lib/src/index_controller/dump_actor/loaders/mod.rs:13]:
```suggestion
    /// Parses the v1 version of the Desc ranking rules `desc(price)`and returns the field name.
```

File: meilisearch-lib/src/index_controller/dump_actor/loaders/v2.rs [hunk 2]
```diff
@@ -1,52 +1,393 @@
...
+        let meta = match meta {
+            compat::UpdateMeta::DocumentsAddition {
+                method,
+                primary_key,
+                ..
+            } => {
+                Update::DocumentAddition {
+                    primary_key,
+                    method,
+                    // jsut ignore if the uuid is no present. If it is needed later, an error will
```
Comment (irevoire) [inline meilisearch-lib/src/index_controller/dump_actor/loaders/v2.rs:201]:
```suggestion
                    // just ignore if the uuid is no present. If it is needed later, an error will
```

File: meilisearch-lib/src/index_controller/update_file_store.rs [hunk 3]
```diff
@@ -73,6 +73,11 @@ impl UpdateFileStore {
         let src_update_files_path = src.as_ref().join(UPDATE_FILES_PATH);
         let dst_update_files_path = dst.as_ref().join(UPDATE_FILES_PATH);
 
+        // No update files to load
+        if !src_update_files_path.exists() {
+            return Ok(());
+        }
+
```
Comment (irevoire) [inline meilisearch-lib/src/index_controller/update_file_store.rs:80]: Is this going to fix #1701?

Comment (irevoire) [reply meilisearch-lib/src/index_controller/update_file_store.rs:80]: This is fixed for the empty dump v1

**Review by curquiza — APPROVED** [review]
I tested, it works on my side 💪

**Review by irevoire — APPROVED** [review]
Thank you!
bors merge

**Comment by irevoire** [comment]
bors merge

**Review by irevoire — APPROVED** [review]
bors merge
bad bors :angry:

**Pull request merged**